### ISSUE TITLE
Receive Timeout causes NullPointerException within internals

### ISSUE THREAD

**Issue by lolboxen** [akka/akka#32015]
Under specific timing conditions, calling `context.cancelReceiveTimeout()` will induce a `NullPointerException` thrown with this stack trace
```
java.lang.NullPointerException: null
	at akka.actor.typed.internal.InterceptorImpl.receive(InterceptorImpl.scala:84)
	at akka.actor.typed.Behavior$.interpret(Behavior.scala:282)
	at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:238)
	at akka.actor.typed.internal.adapter.ActorAdapter.handleMessage(ActorAdapter.scala:128)
	at akka.actor.typed.internal.adapter.ActorAdapter.aroundReceive(ActorAdapter.scala:93)
	at akka.actor.ActorCell.receiveMessage$$$capture(ActorCell.scala:579)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala)
	at akka.actor.ActorCell.invoke(ActorCell.scala:547)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
	at java.base/java.util.concurrent.ForkJoinTask.doExec$$$capture(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

It is my believe the source of this issue is [here](https://github.com/akka/akka/blob/main/akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorContextAdapter.scala#L126). The message to be received is nulled out but the `akka.actor.ReceiveTimeout` is still sitting in the mailbox. When the message is then processed by `InterceptorImpl.scala:84` it attempts to call `getClass` on a null object.

This issue was observed with 2.8.3 and 2.6.21.

We have switched to using `Behaviors.withTimers` to get around this issue per the guarantees it makes on cancellations.

[reproducible evidence](https://github.com/lolboxen/akka-receive-timeout-bug)

**Comment by He-Pin** [comment]
Would you like to provide a PR？

**Comment by octonato** [comment]
fixed in #32016
